test/integration_test/kubevirt_test.go

```diff
@@ -97,54 +99,64 @@ func validateVM(t *testing.T, virtualMachine kubevirtv1.VirtualMachine, vmName s
+func createTemplatePVC(t *testing.T) error {
+	if templatesExist(t) {
+		logrus.Infof("No new PVCs were created for templates. Won't wait for templates to be created.")
+		return nil
 	}
-	waitForImporterPodStart := func() (interface{}, bool, error) {
-		allPods, err := core.Instance().GetPods(kubevirtTemplateNamespace, nil)
-		if err != nil || len(allPods.Items) == 0 {
-			return nil, true, fmt.Errorf("no importer pod not started yet. Retrying")
-		}
-		for _, pod := range allPods.Items {
-			if strings.Contains(pod.Name, importerPodPrefix) {
-				logrus.Infof("importer pod found: %s. Status: %s", pod.Name, pod.Status.Phase)
-				return nil, false, nil
+
+	waitForCompletedAnnotations := func() (interface{}, bool, error) {
+		// Loop through all PVCs and check for annotations that signify existing downloaded templates
+		pvcTemplates, _ := core.Instance().GetPersistentVolumeClaims(kubevirtTemplateNamespace, nil)
+		for _, pvc := range pvcTemplates.Items {
+			storageCondition, _ := pvc.ObjectMeta.Annotations[kubevirtCdiStorageConditionAnnotation]
+			if storageCondition != "Completed" {
+				return nil, true, fmt.Errorf("storage condition is not completed on pvc %s. Status: %s. Retrying.",
+					pvc.Name, pvc.ObjectMeta.Annotations[kubevirtCdiStorageConditionAnnotation])
+			}
+			podPhase, _ := pvc.ObjectMeta.Annotations[kubevirtCdiStoragePodPhaseAnnotation]
+			if podPhase != "Succeeded" {
+				return nil, true, fmt.Errorf("pod phase has not succeeded on pvc %s. Phase: %s. Retrying.",
+					pvc.Name, pvc.ObjectMeta.Annotations[kubevirtCdiStoragePodPhaseAnnotation])
 			}
 		}
-		return nil, false, nil
+		return "", false, nil
 	}
-	_, err = task.DoRetryWithTimeout(waitForImporterPodStart, importerPodStartTimeout, importerPodRetryInterval)
+	_, err := task.DoRetryWithTimeout(waitForCompletedAnnotations, importerPodCompletionTimeout, importerPodRetryInterval)
 	if err != nil {
 		return err
 	}
 
-	// Importer pod has begun, now wait for it to get completed and to disappear
-	waitForImporterPodCompletion := func() (interface{}, bool, error) {
-		allPods, err := core.Instance().GetPods(kubevirtTemplateNamespace, nil)
-		if err != nil {
-			return "", false, nil
+	return err
+}
+
+func templatesExist(t *testing.T) bool {
+	_, err := schedulerDriver.Schedule("",
+		scheduler.ScheduleOptions{
+			AppKeys:   []string{"kubevirt-templates"},
+			Namespace: kubevirtTemplateNamespace,
+		})
+	require.NoErrorf(t, err, "error deploying kubevirt templates")
+
+	pvcTemplates, err := core.Instance().GetPersistentVolumeClaims(kubevirtTemplateNamespace, nil)
+	if err != nil {
+		logrus.Errorf("failed to get PVCs from %s namespace post templates", kubevirtTemplateNamespace)
+		return false
+	}
+
+	// Loop through all PVCs and check for annotations that signify existing downloaded templates
+	for _, pvc := range pvcTemplates.Items {
+		storageCondition, _ := pvc.ObjectMeta.Annotations[kubevirtCdiStorageConditionAnnotation]
+		if storageCondition != "Completed" {
```

Comment:
There is no need to duplicate these checks in this function. We can just call schedulerDriver.Schedule and return. In fact, we can just get rid of the whole function and move `schedulerDriver.Schedule` call to the createTemplatePVC function?